### DISCUSSION TITLE
What event is triggered for turning a Draft PR into a PR?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#23427] [Pull Requests]
<p>Hopefully a simple question: What event is triggered when you turn a Draft PR into a PR? Is there even any event fired? And if so, will it be visible in <a href="https://api.github.com/users/:user/events/public" rel="noopener nofollow ugc">https://api.github.com/users/:user/events/public</a>?</p>
<p>I participate in a GitHub Action to display the latest activities you made on GitHub and this question came up, so I wanted to ask here in the hope that there is a good answer.</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hey <a class="mention" href="/Andre601">@Andre601</a> ! I think there is a webhook that fires for that:<br>
<a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request</a></p>
<p>Check out the <code>action</code> key in the payload, it has support for state changes between draft and non-draft states.</p>

**Reply by eanorige** [reply]
Specifically, the `ready_for_review` action fires when a draft becomes non-draft.

**Comment by Andre601** [comment]
<p>If I understand this right would the <code>action</code> field display <code>converted_to_draft</code> when you convert to a draft and <code>ready_for_review</code> for when it becomes a normal PR?</p>
<p>And I assume this is a normal PullRequestEvent within the public API (Linked in my OP)?</p>

**Comment by Andre601** [comment]
<p>Looking at this further does this solution unfortunately not work for the (public) events endpoint as this kind of action nor the event associated with it seems to be posted there.</p>

**Comment by mpboom** [comment]
<p>Oh no <a class="mention" href="/Andre601">@Andre601</a> it indeed seems my first response was a little short-sighted.</p>
<p>No, I don’t think your use case is possible. It might be a great idea for a <a href="https://support.github.com/contact/feedback" rel="noopener nofollow ugc">feature request</a> though. And who knows, maybe someone else around here knows a workaround.</p>
